### PR TITLE
View + Lifecycle, Plugin approach

### PR DESCRIPTION
**Pull request by CherryPerry**

### CODE REVIEW

File: libraries/rib-base/src/main/java/com/badoo/ribs/core/view/RibView.kt [hunk 1]
```diff
@@ -41,5 +42,9 @@ interface RibView {
      * Detaches a child from this view. Acts as the opposite of [attachChild].
      */
     fun detachChild(child: Node<*>, subtreeOf: Node<*>)
+
+    interface LifecycleAware {
```
Comment (zsoltk) [inline libraries/rib-base/src/main/java/com/badoo/ribs/core/view/RibView.kt:46]: This one doesn't extend Plugin?

File: libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt [hunk 2]
```diff
@@ -160,8 +160,10 @@ open class Node<V : RibView> @VisibleForTesting internal constructor(
             view?.let { view ->
                 view.androidView.restoreHierarchyState(savedViewState)
                 lifecycleManager.onViewCreated()
+                val lifecycle = lifecycleManager.viewLifecycle!!.lifecycle
+                (view as? RibView.LifecycleAware)?.onCreate(lifecycle)
```
Comment (zsoltk) [inline libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt:164]: Shouldn't we here just add it to list of plugins, and notify all plugins as a next step?

File: libraries/rib-base/src/main/java/com/badoo/ribs/core/view/RibView.kt (hunk 1 above)
Comment (CherryPerry) [reply libraries/rib-base/src/main/java/com/badoo/ribs/core/view/RibView.kt:46]: This one should not be confused with ViewAware, which is actual plugin.
This one is more like extension for view implementation.

File: libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt (hunk 2 above)
Comment (CherryPerry) [reply libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt:164]: Do not want to make list of Plugins mutable. If we add the view to plugin list, then we also need to remove it when destroyed.

**Pull request closed without merging**